extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -65,331 +72,381 @@ export const RTC_ERROR_ENUM = {
- * example.com/example.php
- * @param {string} url
- * @return {string}
- * @visibleForTesting
- */
-export function getCalloutParam_(url) {
-  const parsedUrl = parseUrlDeprecated(url);
-  return (parsedUrl.hostname + parsedUrl.pathname).substr(0, 50);
-}
-
-/**
- * For a given A4A Element, sends out Real Time Config requests to
- * any urls or vendors specified by the publisher.
- * @param {!./amp-a4a.AmpA4A} a4aElement
- * @param {!Object<string, !../../../src/service/variable-source.AsyncResolverDef>} customMacros The ad-network specified macro
- *   substitutions available to use.
- * @param {?CONSENT_POLICY_STATE} consentState
- * @return {Promise<!Array<!rtcResponseDef>>|undefined}
- * @visibleForTesting
- */
-export function maybeExecuteRealTimeConfig_(
-  a4aElement, customMacros, consentState) {
-  // TODO(keithwrightbos) - allow pub to override such that some callouts are
-  // still allowed.
-  if (consentState == CONSENT_POLICY_STATE.INSUFFICIENT ||
-      consentState == CONSENT_POLICY_STATE.UNKNOWN) {
-    return;
-  }
-  const rtcConfig = validateRtcConfig_(a4aElement.element);
-  if (!rtcConfig) {
-    return;
+  /**
+   * Converts a URL into its corresponding shortened callout string.
+   * We also truncate to a maximum length of 50 characters.
+   * For instance, if we are passed
+   * "https://example.com/example.php?foo=a&bar=b, then we return
+   * example.com/example.php
+   * @param {string} url
+   * @return {string}
+   */
+  getCalloutParam_(url) {
+    const parsedUrl = Services.urlForDoc(
+        this.a4aElement_.getAmpDoc()).parse(url);
+    return (parsedUrl.hostname + parsedUrl.pathname).substr(0, 50);
   }
-  customMacros['TIMEOUT'] = () => rtcConfig['timeoutMillis'];
-  const promiseArray = [];
-  const seenUrls = {};
-  const rtcStartTime = Date.now();
-  // For each publisher defined URL, inflate the url using the macros,
-  // and send the RTC request.
-  (rtcConfig['urls'] || []).forEach(urlObj => {
-    let url, errorReportingUrl;
-    if (isObject(urlObj)) {
-      url = urlObj['url'];
-      errorReportingUrl = urlObj['errorReportingUrl'];
-    } else if (typeof urlObj == 'string') {
-      url = urlObj;
-    } else {
-      dev().warn(TAG, `Invalid url: ${urlObj}`);
+
+  /**
+   * For a given A4A Element, sends out Real Time Config requests to
+   * any urls or vendors specified by the publisher.
+   * @param {!Object<string, !../../../src/service/variable-source.AsyncResolverDef>} customMacros The ad-network specified macro
+   *   substitutions available to use.
+   * @param {?CONSENT_POLICY_STATE} consentState
+   * @return {Promise<!Array<!rtcResponseDef>>|undefined}
+   * @visibleForTesting
+   */
+  maybeExecuteRealTimeConfig(customMacros, consentState) {
+    // TODO(keithwrightbos) - allow pub to override such that some callouts are
+    // still allowed.
+    if (consentState == CONSENT_POLICY_STATE.INSUFFICIENT ||
+        consentState == CONSENT_POLICY_STATE.UNKNOWN) {
+      return;
```

Comment:
We could send on the ad request how often RTC was surpressed due to consent

Reply:
sounds better suited for follow up rtc refactor than this initial one